### PR TITLE
feat(reports): support benchmark deep links

### PR DESCRIPTION
**Pull request by nahoc**
This PR enables support for deeplinking to certain benchmarks, example:

#### before:
https://reports.risczero.com/benchmarks
(can't link to any specific benchmark)

#### now:
https://reports-and-benchmarks-git-nahoc-bench-deep-link-risczero.vercel.app/benchmarks/Linux-nvidia_rtx_3090_ti
(goes to `Linux NVIDIA RTX 3090 ti`)

https://reports-and-benchmarks-git-nahoc-bench-deep-link-risczero.vercel.app/benchmarks/Linux-g6.xlarge#Linux-g6.xlarge-composite
(scrolls to the `composite` chart)

### CODE REVIEW

**Comment by pdg744** [comment]
Nice! I guess this is just for the "benchmarks" & "application benchmarks" at this stage?

File: web/apps/benchmarks-and-reports/next.config.js
Comment (pdg744) [inline web/apps/benchmarks-and-reports/next.config.js:25]: do you want to confirm this TODO before we land this?

File: web/apps/benchmarks-and-reports/src/app/sitemap.ts
Comment (pdg744) [inline web/apps/benchmarks-and-reports/src/app/sitemap.ts:5]: is this todo still relevant?

**Review by pdg744 — APPROVED** [review]
LGTM! 

I left in-line comments about a couple of TODOs -- will defer to you on whether or not we want to address before we land this

**Review by flaub — APPROVED** [review]

**Pull request merged**
